### ISSUE TITLE
Output depending on formatting of input

### ISSUE THREAD

**Issue by Duvenbeck-SWE** [sqlfluff/sqlfluff#5552]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I tried writing my own configuration file and time after time ran into problems with sqlfluff formatting the statement differently depending on how I format the input.

### Expected Behaviour

As a formatter I expect the output (that is the formatted sql statement) to be the same regardless of what my input formatting is.
Let's take this sql statement as an example:
```sql
select
    name,
    age
from
    mytable
where
    condition1 = 'value1'
    and condition2 = 'value2'
```
It shouldn't matter if I put it in as
```sql
select name, age from mytable where condition1 = 'value1' and condition2 = 'value2'
```
or
```sql
select name, age 
from myTable 
where condition1 = 'value1' 
AND condition2 = 'value2'
```
or
```sql
select name, age 
from myTable 
where condition1 = 'value1' AND condition2 = 'value2'
```
The output should always be the same.
Otherwise where is the reason for establishing rules for sqlfluff within a team/company.

### Observed Behaviour

What happens is that the output very well depends on how the input is formatted. Taking only the default configuration file and not changing anything, the following output can be observed depending on the input:
### Example 1
input:
```sql
select name, age from mytable where condition1 = 'value1' and condition2 = 'value2'
```
output:
```sql
select
    name,
    age
from mytable
where condition1 = 'value1' and condition2 = 'value2'
```
### Example 2
input:
```sql
select name, age 
from myTable 
where condition1 = 'value1' 
AND condition2 = 'value2'
```
output:
```sql
select
    name,
    age
from mytable
where
    condition1 = 'value1'
    and condition2 = 'value2'
```
### Example 3
input:
```sql
select name, age 
from myTable 
where condition1 = 'value1' AND condition2 = 'value2'
```
output:
```sql
select
    name,
    age
from mytable
where condition1 = 'value1' and condition2 = 'value2'
```

### How to reproduce

The problem can be reproduced by using the default configuration and the above mentioned sql statements.
I tried reading from files but also using the stdin option.
Example for stdin:
```
sqlfluff format --dialect tsql -
select name, age
from myTable
where condition1 = 'value1'
AND condition2 = 'value2'
```

### Dialect

tsql

### Version

Python 3.12.0
sqlfluff 2.3.4/2.3.5 (I tried with both)

### Configuration

- use of default configuration file
- cmd command as above

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by naseemx** [comment]
Hey can you assign this to me

**Comment by peterbud** [comment]
Hi @Duvenbeck-SWE , this issue is basically the same as a later submitted #7295

Let me copy the relevant part here from the closure:

While it’s understandable to want total determinism, the philosophy behind `sqlfluff` (and many similar linters) is generally to augment the author's intent rather than to take complete control of the layout. Here are a few points to consider regarding why 'full surrender' of formatting isn't always the goal:

* Contextual Readability: SQL is often deeply nested. Sometimes, an author adds a newline to logically group complex join conditions or filters that a machine might technically fit on one line, but would be harder for a human to scan.

* Minimal Intervention: The goal of a linter is to ensure code meets a specific standard (rules), not necessarily to rewrite it into a singular 'canonical' form. If your code satisfies the max_line_length and indentation rules, the linter views both versions as "legal" expressions of that code.

* Augmentation over Replacement: Think of the linter as a set of guardrails. It prevents you from drifting into "bad" territory (lines too long, inconsistent casing), but it still leaves the "driving" (the specific aesthetic arrangement) to the developer.

Summary of the current behavior:

* In your examples both outputs satisfy the configured rules after formatting. Since `sqlfluff` aims to be a non-destructive tool, it respects the user's choice to be more vertical than the minimum requirement, provided it doesn't violate a specific constraint.

* Achieving the level of determinism you're looking for would require a 'collapse' rule (removing all non-essential whitespace), which is a significant departure from how many SQL teams utilize the tool to maintain readable, hand-tuned queries.
